astrapy/db.py

```diff
@@ -160,6 +234,36 @@ def find_one_and_replace(
 
         return response
 
+    def vector_find_one_and_replace(
+        self,
+        vector,
+        *,
```

Comment:
I suspect it might be argued that `replacement` come before the star, as in `vector_f_o_a_r(vec, new_obj)`. I know it contradicts what I said earlier (lol) but looking at it now it does not seem too much of a stretch

Reply:
I actually think this, and likewise for update, is better too...